### PR TITLE
Add custom elements guide

### PR DESCRIPTION
**Pull request by imprashast**
This solves https://nmp-jira.atlassian.net/browse/WARP-174

### CODE REVIEW

File: docs/guidelines/custom-elements-warp/index.md [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+And finally, run the script to produce a build with references to Lit and Warp Element replaced with references to URLs on the Eik server
+
+```bash
+npm run build
+```
+
+## Publishing components
+
+If you are creating a component or component library that is to be consumed by other teams you will need to create and provide components your components and include instructions on how to setup import mapping.
+
```
Comment (felicia-haggqvist) [inline docs/guidelines/custom-elements-warp/index.md:143]: This sentence is a bit confusing, looks like duplication of components. Maybe you can remove the `your components` part? :)

File: docs/guidelines/custom-elements-warp/index.md [hunk 2]
```diff
@@ -0,0 +1,242 @@
+# Building Custom Elements with Warp
+
```
Comment (felicia-haggqvist) [inline docs/guidelines/custom-elements-warp/index.md:2]: This is probably self-explanatory, but should we emphasize that this is specifically for Web Components and that it is not related to React or Vue? :)

File: docs/guidelines/custom-elements-warp/index.md [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+In order to solve nasty issues like FOUT and FOUC when writing custom elements for Warp, a base class has been created which can be extended to get out of the box, FOUC/FOUT free styling both on the server (SSR) and In the browser (client side).
+
+## How does it work?
+
+The base class, WarpElement, extends PodiumElement which in turn extends LitElement. It’s pretty much the same as extending LitElement, but with styles added in. The base class does the work of detecting whether it is being run server side or client and applies styles differently depending on the which context it detects.
+
+### Server side
+
+On the server side, the base class adds CSS `@import(...)` statements to the server rendered markup for the various Warp CSS files; fonts, brand, resets and components. This works because when the browser sees these statements, it won't render anything until these imports have been resolved so theres no unwanted flicker of unstyled content that you would see if you used this technique client side.
```
Comment (felicia-haggqvist) [inline docs/guidelines/custom-elements-warp/index.md]: Tiny tweak: `theres` should be rewritten to `there's`. :)

File: docs/guidelines/custom-elements-warp/index.md [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+
+## Background
+
+The shadow Dom brings great power but also great complexity, especially once you bring a style library into the mix. 
+
+In order to solve nasty issues like FOUT and FOUC when writing custom elements for Warp, a base class has been created which can be extended to get out of the box, FOUC/FOUT free styling both on the server (SSR) and In the browser (client side).
+
+## How does it work?
+
+The base class, WarpElement, extends PodiumElement which in turn extends LitElement. It’s pretty much the same as extending LitElement, but with styles added in. The base class does the work of detecting whether it is being run server side or client and applies styles differently depending on the which context it detects.
```
Comment (felicia-haggqvist) [inline docs/guidelines/custom-elements-warp/index.md]: `the` in `"depending on the which context it detects.` can be removed. :)

File: docs/guidelines/custom-elements-warp/index.md [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+
+### Performance
+
+All CSS is loaded from absolute URLS on the Eik server. This means we have to pay a cost in terms of additional requests to the Eik server to resolve the CSS files. In addition, a fair amount of CSS needs to be loaded on a cold load. However, theres a really big win here in that since these 4 URLs are unchanging, and since all components that extend the base class will be making fetch calls for the same 4 URLs, the browser is able to all but remove the problem for us with caching. Sure the first cold request using a component or components that use the WarpElement base class will have to make 4 requests to the Eik server, all requests thereafter will not. And this win goes across pages. As users navigate the site, they will encounter more components that make use of the WarpElement base class and the browser will simply serve these files from cache. The WarpElement class itself is also provided via Eik for the same reason and client side, all components should refer to it on the Eik server directly. There is a bit of a waterfall effect where one file loads another which can increase page loads but this can be mitigated with preloading.
```
Comment (felicia-haggqvist) [inline docs/guidelines/custom-elements-warp/index.md]: Same as previous comment regarding `theres` :)

**Review by felicia-haggqvist — APPROVED** [review]

File: docs/guidelines/custom-elements-warp/index.md (hunk 2 above)
Comment (imprashast) [reply docs/guidelines/custom-elements-warp/index.md:2]: Yeah custom elements === web components
